airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessage.kt

```diff
@@ -200,7 +205,62 @@ data class DestinationRecordRaw(
     }
 
     fun asEnrichedDestinationRecordAirbyteValue(): EnrichedDestinationRecordAirbyteValue {
-        TODO()
+        val rawJson = asRawJson()
+
+        // Get the set of field names defined in the schema
+        val schemaFields =
+            when (schema) {
+                is ObjectType -> schema.properties.keys
```

Comment:
nit: if you just did `schema.properties` here, then you could reuse schemaFields in the `(schema as ObjectType).properties` call below